### PR TITLE
PUB-1625 Added custom logging for failed publications

### PR DESCRIPTION
**Pull request by joshblackmoor**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/PUB-1625

### Change description ###

Added custom logging for when a publications fails, this can store up to 32kb of a payload in the message over the existing 8k.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/pip/data/management/service/ValidationService.java [hunk 1]
```diff
@@ -146,24 +150,29 @@ public void validateBody(MultipartFile file) {
      *
      * @param jsonPayload The JSON body to validate.
      */
-    public void validateBody(String jsonPayload, ListType listType) {
+    public void validateBody(String jsonPayload, HeaderGroup headers) {
+        Map<String, String> propertiesMap = headers.getAppInsightsHeaderMap();
```
Comment (KianKwa) [inline src/main/java/uk/gov/hmcts/reform/pip/data/management/service/ValidationService.java:154]: Nothing wrong with the way you construct the headerGroup map, but there is cleaner way to do that. 

If you want to convert the key of your map to upper case, can try the guava library which is good with this type of case conversion:

CaseFormat.LOWER_CAMEL.to(CaseFormat.UPPER_UNDERSCORE, {{ReplaceWithStringToConvert}})

I think the following might be all you need. It is you which way you prefer:

        ObjectMapper mapper = new ObjectMapper();
        mapper.findAndRegisterModules();
        Map<String, Object> map = mapper.convertValue(headerGroup, new TypeReference<>() {});
        Map<String, String> newMap = map.entrySet()
            .stream()
            .filter(e -> e.getValue() != null)
            .collect(Collectors.toMap(
                e -> CaseFormat.LOWER_CAMEL.to(CaseFormat.UPPER_UNDERSCORE, e.getKey()),
                e -> e.getValue().toString()
            ));

File: src/main/java/uk/gov/hmcts/reform/pip/data/management/models/publication/HeaderGroup.java [hunk 2]
```diff
@@ -83,4 +85,23 @@ public class HeaderGroup {
...
+     * @return A map of headers
+     */
+    public Map<String, String> getAppInsightsHeaderMap() {
+        Map<String, String> map = new ConcurrentHashMap<>();
+        map.computeIfAbsent("PROVENANCE", val -> provenance);
+        map.computeIfAbsent("SOURCE_ARTEFACT_ID", val -> sourceArtefactId);
+        map.computeIfAbsent("TYPE", val -> type.toString());
+        map.computeIfAbsent("SENSITIVITY", val -> sensitivity.toString());
+        map.computeIfAbsent("LANGUAGE", val -> language.toString());
+        map.computeIfAbsent("DISPLAY_FROM", val -> displayFrom.toString());
```
Comment (KianKwa) [inline src/main/java/uk/gov/hmcts/reform/pip/data/management/models/publication/HeaderGroup.java:99]: The toString() method is going to output the string in the format like below. If that is what you want to show in appInsights then it is fine. If you prefer to output it using a specific date format might be better to use displayFrom.format(DateTimeFormatter.ofPattern({XXX}). This is the same for all date fields.

[2022, 12, 14, 17, 30, 00, 000000]

Comment (joshblackmoor) [reply src/main/java/uk/gov/hmcts/reform/pip/data/management/models/publication/HeaderGroup.java:99]: hmm from testing it toString() appears to output it in app insights like: 2022-11-22T12:33:55.1070000Z

File: src/main/java/uk/gov/hmcts/reform/pip/data/management/service/ValidationService.java (hunk 1 above)
Comment (joshblackmoor) [reply src/main/java/uk/gov/hmcts/reform/pip/data/management/service/ValidationService.java:154]: Nice that looks good! Will update the branch and test it :)

File: src/main/java/uk/gov/hmcts/reform/pip/data/management/models/publication/HeaderGroup.java (hunk 2 above)
Comment (KianKwa) [reply src/main/java/uk/gov/hmcts/reform/pip/data/management/models/publication/HeaderGroup.java:99]: Ohh cool that is fine then. Will close this one :)

**Review by KianKwa — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/pip/data/management/service/ValidationService.java (hunk 1 above)
Comment (joshblackmoor) [reply src/main/java/uk/gov/hmcts/reform/pip/data/management/service/ValidationService.java:154]: Discussed on slack and leaving it how it is

**Review by ChrisS1512 — CHANGES_REQUESTED** [review]
It looks like a couple of functional tests are failing on this. Tried re-running but still the same. Once those are sorted happy to approve :)

**Review by ChrisS1512 — APPROVED** [review]

**Pull request merged**